protocol/x/ratelimit/util/capacity.go

```diff
@@ -28,33 +27,22 @@ import (
 // On a high level, `capacity` trends towards `baseline` by `capacity_diff` but does not “cross” it.
 func CalculateNewCapacityList(
 	bigTvl *big.Int,
-	limitParams types.LimitParams,
-	prevCapacityList []dtypes.SerializableInt,
+	limiterCapacityList []types.LimiterCapacity,
 	timeSinceLastBlock time.Duration,
 ) (
 	newCapacityList []dtypes.SerializableInt,
-	err error,
 ) {
 	// Declare new capacity list to be populated.
-	newCapacityList = make([]dtypes.SerializableInt, len(prevCapacityList))
+	newCapacityList = make([]dtypes.SerializableInt, len(limiterCapacityList))
 
-	if len(limitParams.Limiters) != len(prevCapacityList) {
```

Comment:
This function now takes in list of `LimiterCapacity`, each of which is a tuple, so no need to throw error here